### ISSUE TITLE
CA 1508 incorrect behavior with pattern match on 'value is > 0'

### ISSUE THREAD

**Issue by Treit** [dotnet/roslyn-analyzers#7729]
The following program triggers CA1508:

```cs
object value = 1.5m;
bool test = value is > 0;
Console.WriteLine(test);
```

This gives:
```
warning CA1508: 'value is > 0' is always 'true'. Remove or refactor the condition(s) to avoid dead code. (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1508)
```

However, the output of this program is actually `false`.

On the other hand, changing the program to this:

```cs
decimal value = 1.5m;
bool test = value is > 0;
Console.WriteLine(test);
```
...actually makes the CA1508 warning go away, even though now the program does indeed always output `true`.

This behavior seems backwards.
